src/test/regress/expected/multi_alter_table_add_constraints.out

```diff
@@ -0,0 +1,459 @@
+ALTER TABLE unique_test_table ADD CONSTRAINT unn_id UNIQUE(id);
+NOTICE:  using one-phase commit for distributed DDL commands
+HINT:  You can enable two-phase commit for extra safety with: SET citus.multi_shard_commit_protocol TO '2pc'
+-- Error out from the related shard. Table can not have two rows with same id.
+INSERT INTO unique_test_table VALUES(1, 'Ahmet');
+INSERT INTO unique_test_table VALUES(1, 'Mehmet');
+ERROR:  duplicate key value violates unique constraint "unn_id_1400001"
+DETAIL:  Key (id)=(1) already exists.
+CONTEXT:  while executing command on localhost:57638
+ALTER TABLE unique_test_table DROP CONSTRAINT unn_id;
+-- Can create unique constraint over multiple columns including
+-- distribution column
+INSERT INTO unique_test_table VALUES(1, 'Mehmet');
+ALTER TABLE unique_test_table ADD CONSTRAINT unn_id UNIQUE(id);
+ERROR:  could not create unique index "unn_id_1400001"
+DETAIL:  Key (id)=(1) is duplicated.
+CONTEXT:  while executing command on localhost:57637
+ALTER TABLE unique_test_table ADD CONSTRAINT unn_id_name UNIQUE(id, name);
+INSERT INTO unique_test_table VALUES(1, 'Mehmet');
+ERROR:  duplicate key value violates unique constraint "unn_id_name_1400001"
+DETAIL:  Key (id, name)=(1, Mehmet) already exists.
+CONTEXT:  while executing command on localhost:57638
+DROP TABLE unique_test_table;
+-- Check "UNIQUE CONSTRAINT" with reference table
+CREATE TABLE unique_test_table_ref(id int, name varchar(20));
+SELECT create_reference_table('unique_test_table_ref');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+-- We can add unique constraint on any column with reference tables
+ALTER TABLE unique_test_table_ref ADD CONSTRAINT unn_name UNIQUE(name);
+ALTER TABLE unique_test_table_ref ADD CONSTRAINT unn_id UNIQUE(id);
+-- Error out. Since the table can not have two rows with the same id. 
+INSERT INTO unique_test_table_ref VALUES(1, 'Ahmet');
+INSERT INTO unique_test_table_ref VALUES(1, 'Mehmet');
+ERROR:  duplicate key value violates unique constraint "unn_id_1400032"
+DETAIL:  Key (id)=(1) already exists.
+CONTEXT:  while executing command on localhost:57637
+-- We can add unique constraint with multiple columns 
+ALTER TABLE unique_test_table_ref DROP CONSTRAINT unn_id;
+ALTER TABLE unique_test_table_ref ADD CONSTRAINT unn_id_name UNIQUE(id,name);
+INSERT INTO unique_test_table_ref VALUES(1, 'Mehmet');
+DROP TABLE unique_test_table_ref;
+-- Check "UNIQUE CONSTRAINT" with append table
+CREATE TABLE unique_test_table_append(id int, name varchar(20));
+SELECT create_distributed_table('unique_test_table_append', 'id', 'append');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- We can add unique constraint on any column with reference tables
+ALTER TABLE unique_test_table_append ADD CONSTRAINT unn_name UNIQUE(name);
+WARNING:  table "unique_test_table_append" has a UNIQUE or EXCLUDE constraint
```

Comment:
We discussed about this with @marcocitus. We want to prevent this behavior but also we do not want to break existing environments. If user has a append distributed table with unique constraint, we should not error out when creating new shard for this table (with COPY for example). 

It seems there are four case to consider. Correct behavior for each case should be;
- Table is not distributed yet, user creates an unique constraint, then distributes by append -> **error out during distributed table creation**
- Table is distributed(before this PR) by append and has NO unique constraint, user creates unique constraint with ALTER TABLE -> **error out during ALTER TABLE**
- Table is distributed(before this PR) by append and has unique constraint, user replicates a shard -> **propagate unique constraints to new shards**
- Table is distributed(before this PR) by append and has unique constraint, user creates new shard -> **propagate unique constraints to new shards**

@metdos we can implement this behavior with one line fix, do you have any objection adding the fix to this PR.